lib/benchpark/cmd/experiment.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2023 Lawrence Livermore National Security, LLC and other
+# Benchpark Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import shutil
+import sys
+
+import benchpark.experiment
+import benchpark.spec
+
+
+def experiment_init(args):
+    experiment_spec = benchpark.spec.ExperimentSpec(" ".join(args.spec))
+    experiment_spec = experiment_spec.concretize()
+
+    experiment = experiment_spec.experiment
+    experiment.initialize()
+
+    if args.basedir:
+        base = args.basedir
+        sysdir = experiment.experiment_id()
+        destdir = os.path.join(base, sysdir)
+    elif args.dest:
+        destdir = args.dest
+    else:
+        raise ValueError("Must specify one of: --dest, --basedir")
+
+    try:
+        os.mkdir(destdir)
+        experiment.generate_description(destdir)
+    except FileExistsError:
+        print(f"Abort: experiment description dir already exists ({destdir})")
+        sys.exit(1)
+    except Exception:
+        # If there was a failure, remove any partially-generated resources
+        shutil.rmtree(destdir)
+        raise
+
+
+def experiment_list(args):
+    raise NotImplementedError("'benchpark experiment list' is not available")
```

Comment:
```suggestion
    experiments = benchpark.repo.all_object_names(object_type="experiments").
    print(experiments)
```
We'll probably want prettier printing at some point, but that's the gist of it.